app/src/main/java/ayds/winchester/songinfo/home/view/SongDateFormatImpl.kt

```diff
@@ -0,0 +1,73 @@
+package ayds.winchester.songinfo.home.view
+
+import ayds.winchester.songinfo.home.model.entities.Song
+
+interface SongDateFormat {
+    fun getFormattedDate(song : Song) : String
+}
+
+internal class SongDateFormatImpl : SongDateFormat {
+    override fun getFormattedDate(song : Song) : String {
+        val date = song.releaseDate
+        val releaseDatePrecision = song.releaseDatePrecision
+        return formatterDate(date, releaseDatePrecision)
+    }
+
+    private fun formatterDate(date : String, releaseDatePrecision : String) : String {
+        var dateFormatted = ""
+        when (releaseDatePrecision) {
+            "day" -> dateFormatted = dayFormat(date)
+            "month" -> dateFormatted = monthFormat(date)
+            "year" -> dateFormatted = yearFormat(date)
+        }
+        return dateFormatted
+    }
+
+    private fun leapYear(year : Int) : Boolean {
+        var isLeapYear = false
+        if ((year % 4 == 0) && (year % 100 != 0 || year % 400 == 0))
+            isLeapYear = true
+        return isLeapYear
+    }
```

Comment:
Se puede simplificar
```suggestion
    private fun leapYear(year : Int) = ((year % 4 == 0) && (year % 100 != 0 || year % 400 == 0))
```

Reply:
Se simplificó el metodo adoptando una sintaxis mas acorde a Kotlin